app/assets/javascripts/initializers/initializeBaseTracking.js.erb

```diff
@@ -25,9 +31,48 @@ function initializeBaseTracking() {
     }, 25);
     eventListening();
   }
-  trackCustomImpressions();
 }
 
+function trackGoogleAnalytics4() {
+  var wait = 0;
+  var addedGA4 = false;
+  var ga4MeasurementCode = document.body.dataset.ga4TrackingId;
+  if (ga4MeasurementCode) {
+    var waitingOnGA4 = setInterval(function() {
+      if (!addedGA4) {
+        <%# Dynamically add the Google Analytics 4 script tag %>
+        var script = document.createElement('script');
+        script.src = "http://www.googletagmanager.com/gtag/js?id=" + ga4MeasurementCode;
+        script.async = true;
+        document.head.appendChild(script);
+      }
+      addedGA4 = true;
+      wait++;
+      if (window.google_tag_manager) {
+        <%# Define the gtag function and call it. Adapted from https://stackoverflow.com/questions/22716542/google-analytics-code-explanation %>
+        window.dataLayer = window.dataLayer || [];
+        function gtag(){dataLayer.push(arguments);}
+
+        window['gtag'] = window['gtag'] || function () {
+          dataLayer.push(arguments)
+        }
+
+        gtag('js', new Date());
+        gtag('config', ga4MeasurementCode, { 'anonymize_ip': true });
+        clearInterval(waitingOnGA4);
+      }
+      if (wait > 85) {
+        clearInterval(waitingOnGA4);
+        <%# The gem we're using server-side (Staccato) is not yet compatible with the Google Analytics 4 tracking code.
+        More details: https://github.com/tpitale/staccato/issues/97 %>
+        <%# fallbackActivityRecording(); %>
```

Comment:
Definitely curious what @jeremyf thinks about this!

Reply:
Mostly I was wondering about the use of 'fallback' tracking. I think erroneously attributed the method to you based on some git history!